xarray/core/weighted.py

```diff
@@ -154,7 +154,8 @@ def _sum_of_weights(
         """ Calculate the sum of weights, accounting for missing values """
 
         # we need to mask data values that are nan; else the weights are wrong
-        mask = da.notnull()
+        # no type checking because ops are injected
+        mask = da.notnull()  # type: ignore
```

Comment:
Does this mean that any library downstream that runs `.notnull` will get a type error? I worry that'll cause a lot of false positives.

What do others think?

Reply:
Nix this! @keewis has handled, too fast for me https://github.com/pydata/xarray/pull/4618